src/main/java/hobbylist/logic/commands/EditCommand.java

```diff
@@ -90,9 +91,14 @@ public CommandResult execute(Model model) throws CommandException {
             throw new CommandException(MESSAGE_DUPLICATE_ACTIVITY);
         }
 
-        model.setActivity(activityToEdit, editedActivity);
-        model.updateFilteredActivityList(Model.PREDICATE_SHOW_ALL_ACTIVITIES);
-        return new CommandResult(String.format(MESSAGE_EDIT_ACTIVITY_SUCCESS, editedActivity));
+        if (!activityToEdit.equals(editedActivity)) {
+            model.setActivity(activityToEdit, editedActivity);
+            model.updateFilteredActivityList(Model.PREDICATE_SHOW_ALL_ACTIVITIES);
```

Comment:
```suggestion

```
At the moment, after editing an activity, the list will display all the activities. I think it might be a good idea if we only update the activity we need and keep the displayed list as it was before editing. What do you think?

Reply:
I'm not sure if it is considered a design change, but I guess since we did not write anything on the UG about it it might be fine? 
This also might not make sense. For example, we `find test` and an activity with name "test" shows up. After editing the name of the activity to "another", it will not show up in the list anymore, which might be kind of weird.

Reply:
The example you gave sounds reasonable for displaying all tasks. 
However, what if 
- the activity's index originally is 20 in the whole list
- you find ... then activity pops up at index 1
- you edit 1 ... then the activity goes down to index 20
This will usually be met if you have a list of too many activities.